zebra-consensus/src/block/subsidies.rs

```diff
@@ -0,0 +1,260 @@
+//! Calculation of Block Subsidy, Funding Streams, and Founders’ Reward
+
+use zebra_chain::{
+    block::Height,
+    parameters::{Network, NetworkUpgrade::*},
+};
+
+use crate::parameters::{fs, Params};
+
+fn slow_start_shift() -> Height {
+    Height(Params::SLOW_START_INTERVAL / 2)
+}
+
+fn slow_start_rate() -> u32 {
+    Params::MAX_BLOCK_SUBSIDY / Params::SLOW_START_INTERVAL
+}
+
+fn halving(height: Height, network: Network) -> u32 {
```

Comment:
Consider doing `1 << halving` inside this function, so callers don't have to remember to do it.

Then you could call this function something more specific, like "halving_factor", and return an amount:

```suggestion
fn halving(height: Height, network: Network) -> Amount<NonNegative> {
```